Plugin.pm

```diff
@@ -628,6 +635,164 @@ sub getFeatured {
 	});
 }
 
+sub getHome {
+	my ( $client, $cb, $args, $params ) = @_;
+
+	getAPIHandler($client)->page(sub {
+		my $modules = shift;
+
+		my $items = [ map {
+			{
+				name => $_->{title},
+				type => 'link',
+				url => $_->{type} eq 'HIGHLIGHT_MODULE' ? \&getHighlights : \&getModule,
+				passthrough => [ { module => $_ } ],
+			}
+		} @$modules ];
+
+		$cb->( {
+			items => $items
+		} );
+	}, 'pages/home' );
+}
+
+sub getHighlights {
+	my ( $client, $cb, $args, $params ) = @_;
+
+	my $module = $params->{module};
+	my $items = [];
+
+	foreach my $entry (@{$module->{highlights}}) {
+		my $item;
+		my $title = $entry->{title};
+
+		$entry = $entry->{item};
+
+		# formats are not exactly consistent with the rest of the API. Small changes
+		# but requires a few exceptions. We could change getImageUrl and typeOfItem...
```

Comment:
Yes, please do tweak those... otherwise you spread the image decisions across the code.